atomic_reactor/utils/operator.py

```diff
@@ -87,153 +179,131 @@ def replace_pullspecs(self, replacement_pullspecs):
-        pullspecs = set()
-        if "containerImage" in annotations:
-            pullspecs.add(self._get_pullspec(annotations, "containerImage", log_template))
-        return pullspecs
 
-    def _get_init_container_pullspecs(self):
-        init_containers = self._get_init_containers()
-        log_template = "%(path)s - Found pullspec for initContainer %(name)s: %(pullspec)s"
-        return set(self._get_pullspec(c, "image", log_template) for c in init_containers)
-
-    def _replace_pullspec(self, obj, key, replacement_pullspecs, log_template=None):
+    def _replace_unnamed_pullspec(self, obj, key, replacement_pullspecs):
         old = ImageName.parse(obj[key])
         new = replacement_pullspecs.get(old)
-        if new is None or new == old:
-            return
-        if log_template is not None:
-            name = obj.get("name") if isinstance(obj, CommentedMap) else None
-            log.debug(log_template,
-                      {"path": self.path, "name": name, "old": old, "new": new})
-        obj[key] = str(new)  # `new` is an ImageName
-
-    def _replace_related_image_pullspecs(self, replacement_pullspecs):
-        related_images = self._get_related_images()
-        log_tmpl = "%(path)s - Replaced pullspec for related image %(name)s: %(old)s -> %(new)s"
-        for i in related_images:
-            self._replace_pullspec(i, "image", replacement_pullspecs, log_tmpl)
-
-    def _replace_env_pullspecs(self, replacement_pullspecs):
-        envs = self._get_related_image_envs()
-        log_template = "%(path)s - Replaced pullspec in %(name)s var: %(old)s -> %(new)s"
-        for e in envs:
-            self._replace_pullspec(e, "value", replacement_pullspecs, log_template)
-
-    def _replace_container_image_pullspecs(self, replacement_pullspecs):
-        containers = self._get_containers()
-        log_template = "%(path)s - Replaced pullspec for container %(name)s: %(old)s -> %(new)s"
-        for c in containers:
-            self._replace_pullspec(c, "image", replacement_pullspecs, log_template)
-
-    def _replace_annotation_pullspecs(self, replacement_pullspecs):
-        annotations = self._get_annotations()
-        log_tmpl = "%(path)s - Replaced pullspec in annotations: %(old)s -> %(new)s"
-        if "containerImage" in annotations:
-            self._replace_pullspec(annotations, "containerImage", replacement_pullspecs, log_tmpl)
-
-    def _replace_init_container_pullspecs(self, replacement_pullspecs):
-        init_containers = self._get_init_containers()
-        log_template = (
-            "%(path)s - Replaced pullspec for initContainer %(name)s: %(old)s -> %(new)s"
-        )
-        for c in init_containers:
-            self._replace_pullspec(c, "image", replacement_pullspecs, log_template)
+        if new is not None and new != old:
+            log.debug("%s - Replaced pullspec: %s -> %s", self.path, old, new)
+            obj[key] = str(new)  # `new` is an ImageName
```

Comment:
if you would use new.to_str() it would be more clearer that it is actually ImageName